atomic_reactor/plugins/cachi2_init.py

```diff
@@ -0,0 +1,154 @@
+
+    def run(self) -> Optional[List[Dict[str, Any]]]:
+        if (not self.workflow.conf.allow_multiple_remote_sources
+                and self.multiple_remote_sources_params):
+            raise ValueError('Multiple remote sources are not enabled, '
+                             'use single remote source in container.yaml')
+
+        if not (self.single_remote_source_params or self.multiple_remote_sources_params):
+            self.log.info('Aborting plugin execution: missing remote source configuration')
+            return None
+
+        if self.multiple_remote_sources_params:
+            self.verify_multiple_remote_sources_names_are_unique()
+
+        if self.dependency_replacements:
+            raise ValueError('Dependency replacements are not supported by Cachi2')
+
+        processed_remote_sources = self.process_remote_sources()
+
+        return processed_remote_sources
+
+    def process_remote_sources(self) -> List[Dict[str, Any]]:
+        """Process remote source requests and return information about the processed sources."""
+
+        remote_sources = self.multiple_remote_sources_params
+        if self.single_remote_source_params:
+            remote_sources = [{
+                "name": "",  # name single remote source with generic name
+                "remote_source": self.single_remote_source_params}]
+
+        processed_remote_sources = []
+
+        self.remote_sources_root_path.mkdir()
+
+        for remote_source in remote_sources:
+            # single source doesn't have name, fake it for cachi2_run task
+            source_name = (
+                remote_source["name"] if self.multiple_remote_sources_params
+                else CACHI2_SINGLE_REMOTE_SOURCE_NAME
+            )
+            self.log.info("Initializing remote source %s", source_name)
+            source_path = self.remote_sources_root_path / source_name
+            source_path.mkdir()
+
+            remote_source_data = remote_source["remote_source"]
+
+            self.write_cachi2_pkg_options(
+                remote_source_data,
+                source_path / CACHI2_PKG_OPTIONS_FILE)
+            self.write_cachi2_for_output_dir(
+                source_name,
+                source_path / CACHI2_FOR_OUTPUT_DIR_OPT_FILE)
+
+            source_path_app = source_path / CACHI2_BUILD_APP_DIR
+            source_path_app.mkdir()
+
+            self.clone_remote_source(
+                remote_source_data["repo"],
+                source_path_app,
+                remote_source_data["ref"]
+            )
+
+            processed_remote_sources.append({
+                "source_path": str(source_path),
+                **remote_source,
+            })
+
+        return processed_remote_sources
+
+    def clone_remote_source(self, repo: str, target_dir: Path, commit: str):
+        self.log.debug("Cloning %s at %s into %s", repo, commit, target_dir)
+        clone_git_repo(
+            repo,
+            target_dir,
+            commit
+        )
```

Comment:
I think you're already aware, but this will also need git-submodule handling eventually

Reply:
yes, there is followup story to support git-submodule "pkg-manager"